### DISCUSSION TITLE
How to plot orbitals from fragment orbitals?

### DISCUSSION THREAD

**Discussion by jjgoings** [goodchemistryco/Tangelo#295] [Q&A] [answered]
Curious how to plot the fragment + bath orbitals from a DMET decomposition, like y'all do for the H10 example in the DMET notebook? It would be helpful for selecting frozen core and ensuring the active orbitals are what I expect.

**Comment by alexfleury-sb** [comment, marked as answer]
Hi Joshua! For plotting the DMET orbitals, we use the coefficients embedded in the `dmet_solver.orbitals.localized_mo` and `bath_orb` variables (`dmet_solver` is a `DMETProblemDecomposition` object). Then, we loop through those to get the `cube` files:
```python
from pyscf import tools
for i in range(dmet_solver.orbitals.localized_mo.shape[1]):
    tools.cubegen.orbital(dmet_solver.orbitals.mol_full, f"M{i+1}.cube", dmet_solver.orbitals.localized_mo[:,i])
```
Note: to get the bath orbitals, we can switch `dmet_solver.orbitals.localized_mo` to `bath_orb`. At this moment, the [bath orbitals](https://github.com/goodchemistryco/Tangelo/blob/748d8619e5dda0c6b203ce4b7073f8acb3927443/tangelo/problem_decomposition/dmet/dmet_problem_decomposition.py#L347) are not saved in the `dmet_solver` object (so a little hack is necessary).

If you want specific MOs, the minimum and maximum MO indices for each fragment are listed in the `dmet_solver.orb_list2` attribute. 

For the visualization step, I personally use VMD. Here is a useful [link](https://www.ks.uiuc.edu/Training/Tutorials/vmd-qmvis/qmvis-tutorial.pdf) (in section 2) for the basic steps.
